src/pages/allCharacters/AllCharacters.tsx

```diff
@@ -5,28 +5,26 @@ import style from './AllCharacters.module.scss'
 import {getAllCharacters} from '../../services/Api'
 import { ICharacter } from '../../services/types';
 import Modal from '../../components/Modal/Modal';
+import { someSlice } from '../../store/inputReducer';
+import { useAppDispatch, useAppSelector } from '../../store/hooks';
 
 
 const AllCharacters:React.FC = () => {
-    const [characters, setCharacters] = useState<ICharacter[]>([]);
-    const [isModalVisible, setIsModalVisible] = useState(false);
-    const [character, setCharacter] = useState<ICharacter>();
+    const {isModalVisible, allCharacters} = useAppSelector(state => state.inputReducer);
+    const {fetchAllCharacters} = someSlice.actions;
+    const dispatch = useAppDispatch();
 
-    const searchedCharacter = (charName:string): void => {
-        setCharacter(characters.find(({ name }) => name.toLowerCase() === charName.toLowerCase()));
-    };
-    
     useEffect(() => {
-        getAllCharacters().then(result => setCharacters(result));
+        getAllCharacters().then(result => dispatch( fetchAllCharacters(result) ));
```

Comment:
refactoring needs here.  your dispatch should fetch the data and reducer should set it in the app store